base/math.jl

```diff
@@ -1255,6 +1255,15 @@ function modf(x::T) where T<:IEEEFloat
     return (rx, ix)
 end
 
+@inline function use_power_by_squaring(n::Union{Int32,Int64,Int128,UInt32,UInt64,UInt128})
```

Comment:
is this function really used during bootstrap?

Reply:
At least, when I called `Base.top_set_bit` the bootstrap compiler could not be built.

Reply:
... and  `Base.BitInteger` was not available. Obviously during bootstrap `^` is called.
Nevertheless I will probably simplify this function, because the exact modelling of the expected runtime, depending on the value of `n`, does not yield essential advantage over using something like `nmin <= n <= nmax` with sensible limits.

Reply:
> nmin <= n <= nmax with sensible limits.
agreed that this is the way forward (especially since then you can just use `Integer`